diff --git a/routebuilder.coffee b/routebuilder.coffee
index 51913b2..3159ca9 100644
--- a/routebuilder.coffee
+++ b/routebuilder.coffee
@@ -23,7 +23,7 @@ SERIALIZERS =
     success : (req, res, result) ->
       res.send(result)
     fail    : (req, res, err) ->
-      res.send 500, err.message
+      res.send 500, err.toString()
   json :
     success : (req, res, result) ->
       res.statusCode = if req.method == "POST" then 201 else 200
@@ -31,10 +31,10 @@ SERIALIZERS =
         status : 'success'
         data   : result
     fail : (req, res, err) ->
-      err.statusCode = err.status
+      err.statusCode = err.status || 500
       res.json err.statusCode,
         status : 'error'
-        data   : err.message
+        data   : err.toString()
   file : 
     success : (req, res, result) ->
       fileName = ''
